x-pack/test/ingest_manager_api_integration/config.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import path from 'path';
+
+import { FtrConfigProviderContext } from '@kbn/test/types/ftr';
+import { defineDockerServersConfig } from '@kbn/test';
+
+export default async function ({ readConfigFile }: FtrConfigProviderContext) {
+  const xPackAPITestsConfig = await readConfigFile(require.resolve('../api_integration/config.ts'));
+
+  const registryPort: string | undefined = process.env.INGEST_MANAGEMENT_PACKAGE_REGISTRY_PORT;
+
+  // mount the config file for the package registry as well as
+  // the directory containing additional packages into the container
+  const dockerArgs: string[] = [
+    '-v',
+    `${path.join(
+      path.dirname(__filename),
+      './apis/fixtures/package_registry_config.yml'
+    )}:/registry/config.yml`,
+    '-v',
+    `${path.join(
+      path.dirname(__filename),
+      './apis/fixtures/test_packages'
+    )}:/registry/packages/test-packages`,
+  ];
+
+  return {
+    testFiles: [require.resolve('./apis')],
+    servers: xPackAPITestsConfig.get('servers'),
+    dockerServers: defineDockerServersConfig({
```

Comment:
So for the endpoint functional tests (those that execute from here `x-pack/test/security_solution_endpoint/config.ts`), we would need to add something similar to this (along with an endpoint package fixture), right?

Any chance this setup could be extracted into shared lib/module so that we can just re-use from our end, and thus ensure we're testing with the same setup?

I would prose adding a method to `x-pack/test/common/services/ingest_manager.ts` that would return some of of this setup - example:

```typescript
return {
  async setup() {
    //....
  },

  getPackageRegistryDockerConfig() {
    return {
      registry: {
        enabled: !!registryPort,
        image: 'docker.elastic.co/package-registry/package-registry:PR-539',
        portInContainer: 8080,
        port: registryPort,
        args: [
          '-v',
          `${path.join(
            path.dirname(__filename),
            './apis/fixtures/package_registry_config.yml'
          )}:/registry/config.yml`,
          '-v',
          `${path.join(
            path.dirname(__filename),
            './apis/fixtures/test_packages'
          )}:/registry/packages/test-packages`,
        ],
        waitForLogLine: 'package manifests loaded into memory',
      },
    }
  }
};
```

We could then use the same setup, add in our package in the `args` and enable this in our runs of Front-end functional tests (and maybe even our api integration tests as well).

//cc: @nnamdifrankie ,  @charlie-pichette , @jonathan-buttner FYI

Reply:
I like the idea. Let's add this after this PR has been reviewed and merged?

Reply:
sounds good @skh 👍